wagtail/admin/forms/collections.py

```diff
@@ -18,11 +19,51 @@ class Meta:
         fields = ('restriction_type', 'password', 'groups')
 
 
+class SelectWidget(forms.Select):
+    """
+    Subclass of Django's select widget that allows disabling options.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.disabled_choices = []
+        super().__init__(*args, **kwargs)
+
+    def create_option(self, name, value, *args, **kwargs):
+        option_dict = super().create_option(name, value, *args, **kwargs)
+        if value in self.disabled_choices:
+            option_dict['attrs']['disabled'] = 'disabled'
+        return option_dict
+
+
 class CollectionForm(forms.ModelForm):
+    parent = forms.ChoiceField(
```

Comment:
This is going over my head. Could you describe your suggestion for an audience who doesn't work with this code all the time? I'm not really sure what you're objecting to, or what you're suggesting as a fix.

Reply:
Apologies for my not-so-great description. It's probably better that I just show you what I mean here:

```
from django.db.models import Min
...

class SelectWithDisabledOptions(forms.Select):
    """
    Subclass of Django's select widget that allows disabled options.
    """
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.disabled_values = ()

    def create_option(self, name, value, *args, **kwargs):
        option_dict = super().create_option(name, value, *args, **kwargs)
        if value in self.disabled_values:
            option_dict['attrs']['disabled'] = 'disabled'
        return option_dict


class CollectionChoiceField(forms.ModelChoiceField):
    widget = SelectWithDisabledOptions

    def __init__(self, *args, disabled_queryset=None, **kwargs):
        super().__init__(*args, **kwargs)
        self._indentation_start_depth = None
        self._disabled_queryset = disabled_queryset
        if disabled_queryset is not None:
            self.widget.disabled_values = disabled_queryset.values_list(self.to_field_name or 'pk')
    
    def _get_disabled_queryset(self):
        return self._disabled_queryset
    
    def _set_disabled_queryset(self, queryset):
        self._disabled_queryset = queryset
        if queryset is None:
            self.widget.disabled_values = ()
        else:
            self.widget.disabled_values = queryset.values_list(self.to_field_name or 'pk', flat=True)

    disabled_queryset = property(_get_disabled_queryset, _set_disabled_queryset)
    
    def _set_queryset(self, queryset):
        min_depth = self.queryset.aggregate(Min('depth'))['depth__min']
        if min_depth is None:
            self.indentation_start_depth = None
        else:
            self.indentation_start_depth = min_depth + 1

    def label_from_instance(self, obj):
        return obj.get_indented_name(self.indentation_start_depth, plain_text=True)
            
 
class CollectionForm(forms.ModelForm):
    parent = CollectionChoiceField(
        queryset=Collection.objects.all()
        required=False,
        help_text=_(
            "Select hierarchical position. Note: a collection cannot become a child of itself or one of its "
            "descendants."
        )
    )
    
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)

        if self.instance._state.adding:
            self.initial['parent'] = Collection.get_first_root_node().pk
        else:
            self.initial['parent'] = self.instance.get_parent().pk
            self.fields['parent'].disabled_queryset = self.instance.get_descendants(inclusive=True)
```

Reply:
Ahhh, Now I see what you mean. Makes sense, so I've incorporated that code. 

Though I did need to change the name of a later class in this file, since it was also called `CollectionChoiceField`. Since it's related to Permissions, I called it `CollectionPermissionsChoiceField`.